app/src/main/java/org/mozilla/fenix/components/metrics/ActivationPing.kt

```diff
@@ -0,0 +1,163 @@
+            null
+        } catch (e: NullPointerException) {
+            Logger.debug("ActivationPing - no Google Advertising ID available")
+            null
+        }
+    }
+
+    /**
+     * Get the salt to use for hashing. This is a convenience
+     * function to help with unit tests.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun getHashingSalt(): String = context.applicationInfo.packageName
+
+    /**
+     * Produces an hashed version of the Google Advertising ID.
+     * We want users using more than one of our products to report a different
+     * ID in each of them. This function runs off the main thread and is CPU-bound.
+     *
+     * @return an hashed and salted Google Advertising ID or null if it was not possible
+     *         to get the Google Advertising ID.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal suspend fun getHashedIdentifier(): String? = withContext(Dispatchers.Default) {
+        getAdvertisingID()?.let { unhashedID ->
+            // Add some salt to the ID, before hashing. For this specific use-case, it's ok
+            // to use the same salt value for all the hashes. We want hashes to be stable
+            // within a single product, but we don't want hashes to be the same across different
+            // products (e.g. Fennec vs Fenix).
+            val salt = getHashingSalt()
+
+            // Apply hashing.
+            try {
+                val saltedID = unhashedID + salt
+                val digest = MessageDigest.getInstance("SHA-256")
+                val hashedBytes = digest.digest(saltedID.toByteArray(Charsets.UTF_8))
+                Base64.encodeToString(hashedBytes, Base64.NO_WRAP)
+            } catch (e: NoSuchAlgorithmException) {
+                Logger.debug("ActivationPing - no SHA-256 available")
+                null
+            }
+        }
+    }
+
+    /**
+     * Fills the metrics and triggers the 'activation' ping.
+     * This is a separate function to simplify unit-testing.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun triggerPing() {
+        // Generate the activation_id.
+        ActivationPing.activationId.generateAndSet()
+
+        // TODO: check if when this gets out of scope, the execution terminates.
+        CoroutineScope(Dispatchers.Default).launch {
+            val hashedId = getHashedIdentifier()
+            if (hashedId != null) {
+                Logger.info("ActivationPing - generating ping with the hashed id")
+                // We have a valid, hashed Google Advertising ID. Send
+                // it with the ping but send no "client_id".
+                ActivationPing.identifier.set(hashedId)
+            }
+
+            Logger.info("ActivationPing - generating ping (has `identifier`: ${hashedId != null})")
+            // TODO: change-me
+            Glean.sendPings(listOf("activation")) //, sendClientId=false)
```

Comment:
You know this already, but this will change once https://github.com/mozilla-mobile/android-components/pull/2792 lands